pkg/cmd/openshift-apiserver/openshiftadmission/register.go

```diff
@@ -0,0 +1,85 @@
+package openshiftadmission
+
+import (
+	buildsecretinjector "github.com/openshift/origin/pkg/build/apiserver/admission/secretinjector"
+	buildstrategyrestrictions "github.com/openshift/origin/pkg/build/apiserver/admission/strategyrestrictions"
+	imagepolicyapi "github.com/openshift/origin/pkg/image/apiserver/admission/apis/imagepolicy"
+	"github.com/openshift/origin/pkg/image/apiserver/admission/imagepolicy"
+	imageadmission "github.com/openshift/origin/pkg/image/apiserver/admission/limitrange"
+	projectrequestlimit "github.com/openshift/origin/pkg/project/apiserver/admission/requestlimit"
+	quotaclusterresourcequota "github.com/openshift/origin/pkg/quota/apiserver/admission/clusterresourcequota"
+	schedulerpodnodeconstraints "github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/plugin/namespace/lifecycle"
+	mutatingwebhook "k8s.io/apiserver/pkg/admission/plugin/webhook/mutating"
+	validatingwebhook "k8s.io/apiserver/pkg/admission/plugin/webhook/validating"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/kubernetes/plugin/pkg/admission/gc"
+	"k8s.io/kubernetes/plugin/pkg/admission/resourcequota"
+)
+
+// TODO register this per apiserver or at least per process
+var OriginAdmissionPlugins = admission.NewPlugins()
+
+func init() {
+	RegisterAllAdmissionPlugins(OriginAdmissionPlugins)
+}
+
+// RegisterAllAdmissionPlugins registers all admission plugins
+func RegisterAllAdmissionPlugins(plugins *admission.Plugins) {
+	// register gc protection plugin
+	gc.Register(plugins)
+	resourcequota.Register(plugins)
+
+	genericapiserver.RegisterAllAdmissionPlugins(plugins)
+	RegisterOpenshiftAdmissionPlugins(plugins)
+}
+
+func RegisterOpenshiftAdmissionPlugins(plugins *admission.Plugins) {
+	projectrequestlimit.Register(plugins)
+	buildsecretinjector.Register(plugins)
+	buildstrategyrestrictions.Register(plugins)
+	imageadmission.Register(plugins)
+	schedulerpodnodeconstraints.Register(plugins)
+	imagepolicy.Register(plugins)
+	quotaclusterresourcequota.Register(plugins)
+}
+
+var (
+	// OpenShiftAdmissionPlugins gives the in-order default admission chain for openshift resources.
+	OpenShiftAdmissionPlugins = []string{
+		lifecycle.PluginName,
+		"OwnerReferencesPermissionEnforcement",
```

Comment:
I wonder if we should have additional test for OpenShift admissions to ensure the hardcoded names are in-sync with upsream?

Reply:
> I wonder if we should have additional test for OpenShift admissions to ensure the hardcoded names are in-sync with upsream?

Pretty sure we will hard fail if we put an invalid item in the list